sync/downstream.py

```diff
@@ -181,14 +181,14 @@ def next_action(self) -> DownstreamAction:
         if not self.requires_try:
             return DownstreamAction.ready
         if self.error:
-            return DownstreamAction.manual_fix
+            self.try_rebase()
 
         latest_try_push = self.latest_valid_try_push
         if (latest_try_push and not latest_try_push.taskgroup_id):
             if latest_try_push.status == "open":
                 return DownstreamAction.wait_try
             elif latest_try_push.infra_fail:
-                return DownstreamAction.manual_fix
+                self.try_rebase()
```

Comment:
So the other problem is that if we have a patch that actually breaks the infrastructure we don't want to keep trying to rebase it indefinitely. So we should do something like keep a bit of state on the sync to say "we've tried to automatically rebase this" and if it's already set we should falback to requiring manual fixup.

Reply:
I've added the flag to do it only once